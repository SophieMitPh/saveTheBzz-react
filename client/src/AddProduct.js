import React, { useState } from "react";
import './AddProduct.css';
import productService from './services/productService';

const AddProduct = () => {

	const initialProductState = {
		id: null,
		name: '',
		description: '',
		price: null
	};

	const [product, setProduct] = useState(initialProductState);
	const [submitted, setSubmitted] = useState(false);

	const handleInputChange = event => {
		const { name, value } = event.target;
		setProduct({ ...product, [name]: value })
	};

	const saveProduct = () => {
		var data = {
			name: product.name,
			description: product.description,
			price: product.price,
		}

		productService.newProduct(data)
			.then(response => {
				setProduct({
					id: response.data.id,
					name: response.data.name,
					description: response.data.description,
					price: response.data.price
				});
				setSubmitted(true);
				console.log(response.data);
			})
			.catch(e => {
				console.log(e);
			});
	};

	const newProduct = () => {
		setProduct(initialProductState);
		setSubmitted(false);
	};

	return (
		<div>
		<h2>Add a new product to your webshop!</h2>
			<div className="signUpRectangle">
				<form className="form-container">
					<input type="text" value={product.name} onChange={handleInputChange} placeholder="Product Name" name="name" required/>
						<textarea type="text" value={product.description} onChange={handleInputChange} placeholder="Description" name="description" required/>
						<input value={product.price} onChange={handleInputChange} placeholder="Price" name="price" required />
						<button type="submit" onClick={saveProduct} id="signup-button">ADD THIS PRODUCT</button>
				</form>
			</div>
		</div>
	);
};

export default AddProduct;
