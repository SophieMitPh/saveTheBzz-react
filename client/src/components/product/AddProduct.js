import React, { useState } from "react";
import '../../css/AddProduct.css';
import productService from './../../services/productService';

const AddProduct = (props) => {

	const initialProductState = {
		id: null,
		name: '',
		description: '',
		price: ''
	};

	const [product, setProduct] = useState(initialProductState);

	const handleInputChange = event => {
		const { name, value } = event.target;
		setProduct({ ...product, [name]: value })
	};

	const saveProduct = (e) => {
		e.preventDefault();
		var data = {
			name: product.name,
			description: product.description,
			price: product.price,
		}

		productService.newProduct(data)
			.then(response => {
				setProduct({
					id: response.data.id,
					name: response.data.name,
					description: response.data.description,
					price: response.data.price
				});
				console.log(response.data);
			})
			.catch(e => {
				console.log(e);
			});
		props.history.push('/products/index');
	};

	return (
		<div>
		<h2>Add a new product to your webshop!</h2>
			<div className="signUpRectangle">
				<form onSubmit={(e) => saveProduct(e)} className="form-container">
					<input id="nameValue" type="text" value={product.name} onChange={handleInputChange} placeholder="Product Name" name="name" required/>
						<textarea type="text" value={product.description} onChange={handleInputChange} placeholder="Description" name="description" required/>
						<input value={product.price} onChange={handleInputChange} placeholder="Price" name="price" required />
						<button type="submit" id="signup-button">ADD THIS PRODUCT</button>
				</form>
			</div>
		</div>
	);
};

export default AddProduct;
