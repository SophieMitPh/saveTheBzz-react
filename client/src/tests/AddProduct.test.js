import React from 'react';
import { shallow } from 'enzyme';
import AddProduct from './../components/product/AddProduct';
import {cleanup} from '@testing-library/react';

afterEach(cleanup)

describe('Add product page', () => {
	const container = shallow(<AddProduct />);

	it('should render correctly in "debug" mode', () => {
		expect(container).toMatchSnapshot();
	});

	it('should match the snapshot', () => {
		expect(container.html()).toMatchSnapshot();
	});

	it('should have a name field', () => {
		expect(container.find('input[name="name"]').length).toEqual(1);
	});

	it('should have a description field', () => {
		expect(container.find('textarea[name="description"]').length).toEqual(1);
	});

	it('should have a price field', () => {
		expect(container.find('input[name="price"]').length).toEqual(1);
	});

	it('should have a submit button', () => {
		expect(container.find('button[type="submit"]').length).toEqual(1);
	});
});