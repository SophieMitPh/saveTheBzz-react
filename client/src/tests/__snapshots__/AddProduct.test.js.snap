// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add product page should match the snapshot 1`] = `"<div><h2>Add a new product to your webshop!</h2><div class=\\"signUpRectangle\\"><form class=\\"form-container\\"><input type=\\"text\\" id=\\"nameValue\\" value=\\"\\" placeholder=\\"Product Name\\" name=\\"name\\" required=\\"\\"/><textarea type=\\"text\\" placeholder=\\"Description\\" name=\\"description\\" required=\\"\\"></textarea><input value=\\"\\" placeholder=\\"Price\\" name=\\"price\\" required=\\"\\"/><button type=\\"submit\\" id=\\"signup-button\\">ADD THIS PRODUCT</button></form></div></div>"`;

exports[`Add product page should render correctly in "debug" mode 1`] = `
<div>
  <h2>
    Add a new product to your webshop!
  </h2>
  <div
    className="signUpRectangle"
  >
    <form
      className="form-container"
    >
      <input
        id="nameValue"
        name="name"
        onChange={[Function]}
        placeholder="Product Name"
        required={true}
        type="text"
        value=""
      />
      <textarea
        name="description"
        onChange={[Function]}
        placeholder="Description"
        required={true}
        type="text"
        value=""
      />
      <input
        name="price"
        onChange={[Function]}
        placeholder="Price"
        required={true}
        value=""
      />
      <button
        id="signup-button"
        onClick={[Function]}
        type="submit"
      >
        ADD THIS PRODUCT
      </button>
    </form>
  </div>
</div>
`;
